src/extension/handler/utils.ts

```diff
@@ -0,0 +1,139 @@
+import { workspace, window, Uri } from 'vscode'
+import { dir } from 'tmp-promise'
+
+import { BOOTFILE } from '../constants'
+
+const config = workspace.getConfiguration('runme.checkout')
+const INTERVAL = 500
+const MINIMAL_TIMEOUT = 10 * 1000
+const DEFAULT_START_FILE = 'README.md'
+
+/**
+ * Get the project directory from the settings object.
+ *
+ * If none is set up, the directory will be a temporary directory.
+ *
+ * If the set up directory doesn't exist, it will ask whether to create it and if not
+ * we return `null` meaning we should cancel the checkout operation.
+ *
+ * @returns `vscode.Uri` with target directory to check out the project to
+ *          `null` if the user doesn't want to create the directory, here we should cancel the operation
+ */
+export async function getProjectDir () {
```

Comment:
I see what you mean here with `handler`. perhaps `uri` is a better name. We do have `utils.ts` in a lot of subdirs and it's at the minimum consistent. Anyways, it can be refactored later so I wouldn't dwell on it for too long.

Reply:
This is more a personal coding convention I follow where I put things like `constants.ts`, `utils.ts` or `types.ts` in folder where they have the most relevance. I could also move them into the `uri.ts` directly. Open for suggestions here to come up with a coding style that is convenient for everyone.